### PR TITLE
Hide YouTube from the list of available social networks.

### PR DESCRIPTION
**Pull request by EvanHerman**
@fjarrett @jonathanbardo 

I think it's best to temporarily remove the option to select YouTube until we have a better method for auto-populating the field. Maybe we just populate the field with `https://www.youtube.com/`.

### CODE REVIEW

**Comment by fjarrett** [comment]
@EvanHerman I'm worried here about back compat for those already using it. What if we add a `deprecated => true` field to the data array and add logic to the plugin that will remove it from the UI if it's not currently in use?

File: includes/class-social.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ public function widget( $args, $instance ) {
 
 		foreach ( $fields as $field ) {
 
-			if ( empty( $field['value'] ) || ! $field['show_front_end'] ) {
+			if ( empty( $field['value'] ) || ! $field['show_front_end'] || $this->is_social_media_deprecated( $field ) ) {
```
Comment (fjarrett) [inline includes/class-social.php]: @EvanHerman Won't this always hide a deprecated field, even if its value isn't empty?

```php
if ( empty( $field['value'] ) || ! $field['show_front_end'] || ( ! empty( $field['value'] ) && ! empty( $field['deprecated'] ) ) ) {
```

Comment (fjarrett) [inline includes/class-social.php]: @EvanHerman I don't think we have to make any changes to the front-end widget output. If a field has a value, we should display it.

File: includes/class-social.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function form( $instance ) {
 
 		foreach ( $fields as $key => $field ) {
 
-			if ( ! isset( $field['social'] ) ) {
+			if ( ! isset( $field['social'] ) || $this->is_social_media_deprecated( $field ) ) {
```
Comment (fjarrett) [inline includes/class-social.php]: @EvanHerman I don't think we want to do this either. If the field is currently in use, we should still show it in the widget form.

Comment (fjarrett) [inline includes/class-social.php]: @EvanHerman Oh no sorry, you are right. This is for the icons, I was was mistaken and thought it was for the fields.

File: includes/class-social.php [hunk 3]
```diff
@@ -272,4 +272,19 @@ protected function print_label( array $field ) {
...
+	/**
+	 * Is the social media selection deprecated
+	 *
+	 * @param  array  $field Field array
+	 *
+	 * @return boolean
+	 *
+	 * @since NEXT
+	 */
+	protected function is_social_media_deprecated( $field ) {
```
Comment (fjarrett) [inline includes/class-social.php]: @EvanHerman Having a method for this is overkill IMO.

And `$this->is_social_media_deprecated( $field )` is longer than `! empty( $field['deprecated'] )`.

**Comment by EvanHerman** [comment]
@fjarrett 

This new patch works well with backwards compat.

Users who have YouTube selected with a value will still see the icon and be able to alter the value of the URL.

New users will not have YouTube selectable.

Once the user removes the value from the YouTube URL field, or unselects the YouTube icon and updates the widget the icon will no longer be visible on either the dashboard or the front end of the site.

**Comment by fjarrett** [comment]
Works great!

**Pull request merged**